src/binary/buffer.rs

```diff
@@ -0,0 +1,90 @@
+use crate::common::{Bytes32, Position};
+use core::mem::size_of;
+
+const BUFFER_SIZE: usize = size_of::<Position>() + size_of::<Bytes32>();
+pub type Buffer = [u8; BUFFER_SIZE];
+pub const DEFAULT_BUFFER: &Buffer = &[0; BUFFER_SIZE];
```

Comment:
Would it make sense to just have this as an implementation of `Default` for Buffer?

Reply:
That would be my impulse too, but unfortunately we cannot implement traits on arrays (not even type aliased arrays). The compiler says: `Only traits defined in the current crate can be implemented for arbitrary types`.